File: data/code_pairs/fortran/1261.f90
subroutine intern(j, c, mb, nb, r, dr, nbond)
    integer, intent(in) :: j, nbond
    double precision, dimension(j,4,3), intent(in) :: c
    integer, dimension(nbond), intent(in) :: mb, nb
    double precision, dimension(j,nbond), intent(out) :: r
    double precision, dimension(j,nbond,4,3), intent(out) :: dr
    integer :: i
    double precision :: dx, dy, dz

    do i = 1, nbond
        dx = c(j, mb(i), 1) - c(j, nb(i), 1)
        dy = c(j, mb(i), 2) - c(j, nb(i), 2)
        dz = c(j, mb(i), 3) - c(j, nb(i), 3)
        r(j, i) = sqrt(dx**2 + dy**2 + dz**2)
        dr(j, i, mb(i), 1) = dx / r(j, i)
        dr(j, i, mb(i), 2) = dy / r(j, i)
        dr(j, i, mb(i), 3) = dz / r(j, i)
        dr(j, i, nb(i), 1) = -dx / r(j, i)
        dr(j, i, nb(i), 2) = -dy / r(j, i)
        dr(j, i, nb(i), 3) = -dz / r(j, i)
    end do
end subroutine intern

program test_intern
    implicit none
    integer, parameter :: n=1, nbond=2
    double precision, dimension(n,nbond) :: r
    double precision, dimension(n,nbond,4,3) :: dr
    double precision, dimension(n,4,3) :: c
    integer, dimension(nbond) :: mb, nb

    ! Initialize mb and nb
    mb = (/1, 2/)
    nb = (/3, 4/)

    ! Initialize matrix c
    c(1, 1, 1) = 0.0; c(1, 2, 1) = 1.0; c(1, 3, 1) = 1.0; c(1, 4, 1) = 0.0
    c(1, 3, 2) = 1.0; c(1, 4, 2) = 1.0; c(1, :, 3) = 0.0

    call intern(1, c, mb, nb, r, dr, nbond)

    ! Output results for manual inspection
    print *, 'Fortran results:'
    print *, 'r:', r
    print *, 'dr(mb):', dr(:,:,1,:)
    print *, 'dr(nb):', dr(:,:,2,:)
end program test_intern

File: data/code_pairs/cpp/1261.cpp
#include <vector>
#include <cmath>
#include <iostream>
#include <cassert>

void intern(int j, const std::vector<std::vector<std::vector<double>>>& c,
            const std::vector<int>& mb, const std::vector<int>& nb,
            std::vector<std::vector<double>>& r,
            std::vector<std::vector<std::vector<std::vector<double>>>>& dr,
            int nbond) {
    for (int i = 0; i < nbond; ++i) {
        double dx = c[j][mb[i] - 1][0] - c[j][nb[i] - 1][0];
        double dy = c[j][mb[i] - 1][1] - c[j][nb[i] - 1][1];
        double dz = c[j][mb[i] - 1][2] - c[j][nb[i] - 1][2];
        r[j][i] = sqrt(dx*dx + dy*dy + dz*dz);
        dr[j][i][mb[i] - 1][0] = dx / r[j][i];
        dr[j][i][mb[i] - 1][1] = dy / r[j][i];
        dr[j][i][mb[i] - 1][2] = dz / r[j][i];
        dr[j][i][nb[i] - 1][0] = -dx / r[j][i];
        dr[j][i][nb[i] - 1][1] = -dy / r[j][i];
        dr[j][i][nb[i] - 1][2] = -dz / r[j][i];
    }
}

int main() {
    const int n = 1, nbond = 2;
    std::vector<std::vector<std::vector<double>>> c(n, std::vector<std::vector<double>>(4, std::vector<double>(3, 0.0)));
    std::vector<int> mb = {1, 2}; // Adjusted for 1-based indexing in Fortran
    std::vector<int> nb = {3, 4}; // Same here
    std::vector<std::vector<double>> r(n, std::vector<double>(nbond, 0.0));
    std::vector<std::vector<std::vector<std::vector<double>>>> dr(n, std::vector<std::vector<std::vector<double>>>(nbond, std::vector<std::vector<double>>(4, std::vector<double>(3, 0.0))));

    // Initialize matrix c with some values for testing
    c[0][0][0] = 0.0; c[0][1][0] = 1.0; c[0][2][0] = 1.0; c[0][3][0] = 0.0;
    c[0][2][1] = 1.0; c[0][3][1] = 1.0;

    intern(0, c, mb, nb, r, dr, nbond);

    // Output results for manual inspection
    std::cout << "C++ results:" << std::endl;
    std::cout << "r: ";
    for (const auto& val : r[0]) {
        std::cout << val << " ";
    }
    std::cout << std::endl;

    std::cout << "dr(mb): ";
    for (int i = 0; i < nbond; ++i) {
        for (int k = 0; k < 3; ++k) {
            std::cout << dr[0][i][mb[i] - 1][k] << " ";
        }
    }
    std::cout << std::endl;

    std::cout << "dr(nb): ";
    for (int i = 0; i < nbond; ++i) {
        for (int k = 0; k < 3; ++k) {
            std::cout << dr[0][i][nb[i] - 1][k] << " ";
        }
    }
    std::cout << std::endl;

    return 0;
}